beater/beater.go

```diff
@@ -275,14 +275,18 @@ func (r *reloader) Reload(configs []*reload.ConfigWithMeta) error {
 	if err != nil {
 		return err
 	}
-	var namespace string
-	if integrationConfig.DataStream != nil {
-		namespace = integrationConfig.DataStream.Namespace
-	}
 
 	r.mu.Lock()
-	r.namespace = namespace
 	r.rawConfig = integrationConfig.APMServer
+	// Merge in datastream namespace passed in from apm integration
+	if integrationConfig.DataStream != nil && integrationConfig.DataStream.Namespace != "" {
+		c := agentconfig.MustNewConfigFrom(map[string]interface{}{
+			"data_streams.namespace": integrationConfig.DataStream.Namespace,
+		})
+		if r.rawConfig, err = agentconfig.MergeConfigs(r.rawConfig, c); err != nil {
+			return err
+		}
+	}
```

Comment:
@axw I added the `integrationConfig.DataStream.Namespace != ""` check because, although the integration doesn't allow the user to enter a blank value, it seemed prudent to also check server-side. If you don't think this is necessary I'm happy to remove it.

Reply:
Agreed, seems prudent to me too.